destalinator.py

```diff
@@ -120,7 +120,12 @@ def stale(self, channel_name, days):
         Definition of stale is: no messages in the last `days` which are not from config.ignore_users.
         """
         if not self.channel_minimum_age(channel_name, days):
-            self.logger.debug("Channel #%s is not yet of minimum_age; skipping stale messages check", channel_name)
+            return False
+
+        if self.ignore_channel(channel_name):
+            return False
+
+        if self.slacker.channel_has_only_restricted_members(channel_name):
```

Comment:
These are being checked again _after_ fetching all the messages, but checking before skips fetching all the messages.